Week3/homework/step4-bonus.js

```diff
@@ -1,10 +1,16 @@
 'use strict';
 
-const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
+const names = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 
-// Add your function here. Try and come up with a good name for this function
+const uniq = names
+  .map((name) => {
+    return { count: 1, name: name }
+  })
+  .reduce((a, b) => {
+    a[b.name] = (a[b.name] || 0) + b.count
+    return a;
+  }, {});
 
-// Replace `yourFunction` with the name of the function you just created
-const uniqueValues = yourFunction(values);
+const sorted = Object.keys(uniq).sort((a, b) => uniq[a] < uniq[b])
 
-console.log(uniqueValues);
+console.log(sorted);
```

Comment:
✅ Wow, I love this solution it is the most creative solution to this problem I've seen.  And sorted. Nice 👍 
I was worried about efficiency, but actually, the typical answer that uses `array.includes` does far more traversals.
If you just want to end up with your object in one traversal, although with out the nice count, you can skip the map and just do the reduce.  I'll keep getting the keys as a separate expression for readability.
```
const uniqObj = names.reduce((acc, item) => {
  return {...acc,
          ...{[item]: null}};
}, {});

const uniques = Object.keys(uniqObj);
```
So, very nice work especially using higher-order functions to such effect. 😄